src/Courses/CourseDetailsTabs/StatisticsTab.tsx

```diff
@@ -497,63 +517,68 @@ export const StatisticsTab: React.FC<StatisticsTabProps> = ({ course, userId })
+                            <div className="d-flex">
+                                <h6
+                                    style={{
+                                        whiteSpace: 'nowrap',
+                                        overflow: 'hidden',
+                                        textOverflow: 'ellipsis'
+                                    }}
+                                    className="MuiTypography-root MuiTypography-h6"
                                 >
-                                    {grade.locked ? <><BsLock/> Unlock</>: <><BsUnlock/> Lock</>}
-                                </Button>
-                            </>
-                            }
-                        </div>
-                    )}
-                    columns={(view === StatisticsView.ATTEMPTS || view === StatisticsViewFilter.PROBLEMS_FILTERED) ? attemptCols : gradeCols}
-                    data={rowData}
-                    actions={actions}
-                    onRowClick={nextView}
-                    options={{
-                        exportButton: true,
-                        exportAllData: true,
-                        sorting: true
-                    }}
-                    detailPanel={hasDetailPanel ? [{
-                        icon: () => <ChevronRight />,
-                        render: renderProblemPreview
-                    }] : undefined}
-                    localization={{ header: { actions: '' } }}
-                />
+                                    {getTitle()}
+                                </h6>
+                                {userType === UserRole.PROFESSOR && !_.isNil(userId) && !_.isNil(grade) && (view === StatisticsViewFilter.PROBLEMS_FILTERED) && 
+                                <>
+                                    <Button
+                                        className="ml-3 mr-1"
+                                        onClick={() => setGradesState({
+                                            ...gradesState,
+                                            view: GradesStateView.OVERRIDE
+                                        })}
+                                    >
+                                        <>
+                                            <BsPencilSquare/> Override
+                                        </>
+                                    </Button>
+
+                                    <Button
+                                        variant={grade.locked ? 'warning' : 'danger'}
+                                        className="ml-1 mr-1"
+                                        onClick={() => {
+                                            logger.silly('Stats tab: [table button] setting gradesstate');
+                                            setGradesState({
+                                                ...gradesState,
+                                                view: GradesStateView.LOCK
+                                            });}}
```

Comment:
NIT: IMO we should have all opening closures have closing closures at the same indentation
You can do multiple opening and closing on the same line however if you have 2 openings togeter you should have 2 closing

In this example there is an opening closure for onClick prop, then another one for method start
However on this last line the setGradesState parenthesis and bracket are there as well

It makes it harder to read however if we get a formatter in here it will automically be cleaned up

Reply:
done